homeassistant/components/open_meteo/weather.py

```diff
@@ -55,6 +60,15 @@ def __init__(
             name=entry.title,
         )
 
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        super()._handle_coordinator_update()
+        assert self.platform.config_entry
+        self.platform.config_entry.async_create_task(
```

Comment:
The coordinators already hold the config entry, this feels like a weird route?

Reply:
That's a good point 👍

Reply:
Fixed, `OpenMeteoWeatherEntity` now extends `CoordinatorWeatherEntity`